TW-141 Activities are idempotent (#1767)
temporalio/documentation: docs-src/concepts/what-is-an-activity-definition.md

![Activity Definition](/diagrams/activity-definition.svg)

#### Idempotency

Temporal recommends that Activities be idempotent.

> Activities are idempotent if multiple [Activity Task Executions](/concepts/what-is-an-activity-task-execution) do not change the state of the system beyond the first Activity Task Execution.

Reviewer: ```suggestion
An Activity is idempotent if multiple [Activity Task Executions](/concepts/what-is-an-activity-task-execution) do not change the state of the system beyond the first Activity Task Execution.
```

> Activities may be tried more than once even if the Retry Policy says they shouldn't. Because of that, you may need to use idempotency keys for critical side effects.

Reviewer: ```suggestion
An Activity can be tried more than once even if the Retry Policy says it shouldn't.
Because of that behavior, you might need to use idempotency keys for critical side effects.
```
Capitalize "Side Effects" and link to glossary term?

> The lack of idempotency may affect the correctness of your application, but does not affect Temporal (in other words, it does not lead to a platform error).

Reviewer: ```suggestion
The lack of idempotency might affect the correctness of your application but does not affect Temporal.
(In other words, lack of idempotency does not lead to a platform error.)
```
Should "Temporal" be "the Temporal Platform" in this instance?

> In some cases, whether something is idempotent does not affect the correctness of an application. For example, if you have a monotonically-incrementing counter, you may not care that retries increment the counter because you don’t care about the actual value, only that the current value is greater than a previous value.

Reviewer: ```suggestion
In some cases, whether something is idempotent does not affect the correctness of an application.
For example, if you have a monotonically incrementing counter, you might not care that retries increment the counter because you don’t care about the actual value, only that the current value is greater than a previous value.
```
